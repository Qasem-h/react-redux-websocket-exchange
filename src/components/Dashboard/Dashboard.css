.box {
  margin-bottom: 20px;
  background: #1f1f1f;
  border-radius: 5px;
  box-shadow: 0px 0px 5px 0px rgba(142, 139, 133, 0.32);
  padding: 0px;
  -webkit-transition: 0.5s;
  transition: 0.5s;

  padding: 20px;
}

.ant-table-row td {
  padding: 10px !important;
  font-size: 14px;
}

@media only screen and (max-width: 600px) {
  .ant-table-row td {
    font-size: 12px;
    padding: 5px;
  }
}
