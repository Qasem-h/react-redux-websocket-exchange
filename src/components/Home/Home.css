h1 {
  font-size: 3.7em;
  color: #fff;
  font-weight: 700;
  line-height: 80px;
  margin-bottom: 15px;
}
em {
  font-weight: 400;
  font-size: 35px;
}
.container {
  height: 100vh;
  width: 1250px;
}

.home-img {
  padding-top: 4em;
}

.home-copy {
  padding-top: 7em;
}

.connected {
  font-size: 23px;
  color: #30d9f9;
  font-weight: 400;
  margin-bottom: 15px;
}

.primary {
  background: rgb(240, 185, 11);
  border: 0;
  height: 60px;
  color: rgb(22, 22, 22);
  font-size: 16px;
  text-transform: uppercase;
  width: 220px;
  border-radius: 50px;
  margin-top: 25px;
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  transition: 0.4s;
  outline: none;
  cursor: pointer;
  font-size: 17px;
  font-weight: 600;
}

button:hover {
  background-color: #30d9f9;
}

@media only screen and (max-width: 600px) {
  h1 {
    font-size: 2em;
  }
  em {
    font-size: 30px;
  }
  .connected img {
    display: none;
  }
}
