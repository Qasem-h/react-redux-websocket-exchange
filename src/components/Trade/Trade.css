.trade-title {
  color: #30d9f9;
  font-weight: 400;
  padding-left: 20px;
}
@media only screen and (max-width: 600px) {
  .trade-title {
    font-size: 20px;
    padding-left: 20px;
  }
}
