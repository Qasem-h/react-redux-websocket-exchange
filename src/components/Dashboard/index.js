import React from "react";
import { connect } from "react-redux";
import { Table } from "antd";
import axios from "axios";
import { Link } from "react-router-dom";

import { initializeSocket } from "../../reducers/socket.js";

import "./Dashboard.css";
import Widget from "../Widget";
import WidgetSmall from "../WidgetSmall";
import BinanceChart from "../Chart/BinanceChart";
import neo from "../../assets/img/1024.png";
import monero from "../../assets/img/monero-symbol-1280.png";

const columns = [
  {
    title: "Symbol",
    dataIndex: "s",
    render: (symbol, record) => (
      <Link to={"/trade/" + symbol} style={{ color: "#30d9f9" }}>
        {symbol}
      </Link>
    )
  },
  {
    title: "Open Price",
    dataIndex: "c"
  },
  {
    title: "High Price",
    dataIndex: "h"
  },
  {
    title: "Low Price",
    dataIndex: "l"
  }
];

class Dashboard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      binanceData: [],
      cryptos: [],
      btcHistorical: null
    };
  }
  componentDidMount() {
    this.props.initializeSocket();
    this.fetchMultiSymbols();
    this.fetchHistoricalBTC();
  }

  fetchMultiSymbols = () => {
    axios
      .get(
        "https://min-api.cryptocompare.com/data/pricemulti?fsyms=ETH,BNB,NEO,XMR&tsyms=BTC,USD,EUR,JPY"
      )
      .then(res => {
        const cryptos = res.data;
        console.log("Cryptos", cryptos);
        this.setState({ cryptos: cryptos });
      });
  };

  fetchHistoricalBTC = () => {
    axios
      .get(
        "https://min-api.cryptocompare.com/data/histoday?fsym=BTC&tsym=USD&limit=30&aggregate=3&e=CCCAGG"
      )
      .then(res => {
        const btcHistorical = res.data;
        this.setState({ btcHistorical: btcHistorical.Data });
      });
  };

  render() {
    const btcHistorical = this.state;
    console.log(btcHistorical);
    const alternatingColor = ["rgb(237, 203, 43)", "#30d9f9"];
    const widgetItem = Object.keys(this.state.cryptos).map((key, index) => (
      <Widget
        style={{
          backgroundColor: alternatingColor[index % alternatingColor.length]
        }}
        key={key}
        symbol={key}
        usd={this.state.cryptos[key].USD}
        eur={this.state.cryptos[key].EUR}
        jpy={this.state.cryptos[key].JPY}
      />
    ));
    return (
      <div className="container pt-4 mt-4" data-test="dashboard-display">
        <div className="row">
          <div className="col-md-6">
            <div className="box" data-test="crypto-container">
              <Table
                rowKey={record => record.s}
                columns={columns}
                dataSource={this.props.feed}
                size="small"
                pagination={false}
                style={{ color: "white" }}
              />
            </div>
          </div>
          <div className="col-md-6">
            <BinanceChart charData={btcHistorical} />
            <div className="d-flex flex-wrap justify-content-between">
              {widgetItem}
            </div>
            <div className="row">
              <div className="col-md-6">
                <WidgetSmall
                  url={neo}
                  title="NEO Coin"
                  btcprice="0.04"
                  percentage="1.35%"
                />
              </div>
              <div className="col-md-6">
                <WidgetSmall
                  url={monero}
                  title="Monero"
                  btcprice="0.017"
                  percentage="1.35%"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  feed: state.socket.feed
});

export default connect(
  mapStateToProps,
  { initializeSocket }
)(Dashboard);
