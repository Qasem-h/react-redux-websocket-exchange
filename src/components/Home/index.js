import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { initializeSocket } from "../../reducers/socket.js";
import homeimg from "../../assets/img/bg-binance.png";
import computer from "../../assets/img/happy-computer.png";
import "./Home.css";

class Home extends Component {
  componentDidMount() {
    this.props.initializeSocket();
  }

  render() {
    const { socket } = this.props;
    const online = socket.connected ? (
      <div className="connected">
        <img src={computer} alt="success" style={{ width: "15%" }} /> You are
        now connected to Binance
      </div>
    ) : (
      <div className="connected">Connecting to Binance stream...</div>
    );
    return (
      <div data-test="home-display">
        <div className="container">
          <div className="row">
            <div className="col-md-6 home-copy">
              <h1>
                Cryptocurrency Exchange
                <br />
                <em>Largest coin selection</em>
              </h1>
              {online}
              <div>
                <Link to="/dashboard">
                  <button className="btn primary" data-test="button-display">
                    Start trading
                  </button>
                </Link>
              </div>
            </div>

            <div className="col-md-6 home-img">
              <img src={homeimg} alt="blockchain" style={{ width: "450px" }} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  socket: state.socket
});

export default connect(
  mapStateToProps,
  { initializeSocket }
)(Home);
