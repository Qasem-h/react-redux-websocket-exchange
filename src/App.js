import React, { Component } from "react";
import Nav from "./components/Nav";
import Home from "./components/Home";
import Dashboard from "./components/Dashboard";
import Trade from "./components/Trade";
import { Switch, Route, withRouter } from "react-router-dom";
import Loadable from "react-loadable";

class App extends Component {
  render() {
    return (
      <div>
        <Nav />

        <Switch>
          <Route exact path="/" component={props => <Home />} />
          <Route path="/dashboard" component={props => <Dashboard />} />
          <Route path="/trade/:symbol" component={Trade} />
        </Switch>
      </div>
    );
  }
}

export default withRouter(App);
