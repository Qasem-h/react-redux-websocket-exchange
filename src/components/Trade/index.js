import React, { Fragment } from "react";
import { connect } from "react-redux";
import "./Trade.css";
import { Table } from "antd";
import axios from "axios";
import { initializeSocket } from "../../reducers/socket.js";

const columns = [
  {
    title: "Symbol",
    dataIndex: "s"
  },
  {
    title: "Order Trade Id",
    dataIndex: "a"
  },
  {
    title: "Price",
    dataIndex: "p"
  },
  {
    title: "Quantity",
    dataIndex: "q"
  }
];

class Trade extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      orderData: [],
      loading: true
    };
  }

  async componentWillMount() {
    let symbol = this.props.match.params.symbol.toLowerCase();

    axios({
      method: "get",
      url: `https://cors-anywhere.herokuapp.com/https://www.binance.com/api/v1/aggTrades?limit=100&symbol=${
        this.props.match.params.symbol
      }`
    }).then(data => {
      this.setState({ orderData: data });
    });
    this.setState({ loading: false });
  }

  render() {
    if (!this.state.loading) {
      return <h2>Loading..</h2>;
    }
    console.log(
      "yoooooo",
      this.props.match.params.symbol,
      this.state.orderData
    );
    return (
      <Fragment>
        <div className="container">
          <div className="row pt-4 mt-4">
            <div className="text-center">
              <h2 className="trade-title">
                Detailed Trade History for{" "}
                <b className="text-white">{this.props.match.params.symbol}</b>{" "}
                Pair
              </h2>
            </div>
          </div>
          <div className="row pt-4">
            <div className="col-md-8">
              <div className="box">
                {this.state.loading === false ? (
                  <Table
                    rowKey={record => record.s}
                    columns={columns}
                    dataSource={this.state.data.data}
                    size="small"
                    pagination={false}
                    style={{ color: "white" }}
                  />
                ) : (
                  "Loading"
                )}
              </div>
            </div>
            <div className="col-md-4">two</div>
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  orderBook: state.socket.orderBook
});

export default connect(
  mapStateToProps,
  { initializeSocket }
)(Trade);
