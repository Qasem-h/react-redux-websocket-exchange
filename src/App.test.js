import * as React from "react";
import { shallow } from "enzyme";
import App from "./App";
import Home from "./components/Home";
import Dashboard from "./components/Dashboard";
import Trade from "./components/Trade";

//Not a huge fan of snapshot testing unless its for routes!

describe("App", () => {
  describe("component", () => {
    let element;

    beforeEach(() => {
      element = <App />;
    });
    it("renders as expected", () => {
      const component = shallow(element);
      expect(component).toMatchSnapshot();
    });
    it("routes / to Home component", () => {
      const component = shallow(element);
      expect(
        component
          .find('Route[exact=true][path="/"]')
          .first()
          .prop("component")
      ).toBe(Home);
    });
    it("routes /dashboard to Dashboard", () => {
      const component = shallow(element);
      expect(
        component
          .find('Route[path="/dashboard"]')
          .first()
          .prop("component")
      ).toBe(Dashboard);
    });
    it("routes /trade/:symbol to Trade", () => {
      const component = shallow(element);
      expect(
        component
          .find('Route[path="/trade/:symbol"]')
          .first()
          .prop("component")
      ).toBe(Trade);
    });
  });
});
