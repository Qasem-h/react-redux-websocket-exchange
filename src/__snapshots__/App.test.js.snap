// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <_default />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/dashboard"
          />
          <Route
            component={[Function]}
            path="/trade/:symbol"
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/dashboard"
            />,
            <Route
              component={[Function]}
              path="/trade/:symbol"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/trade/:symbol",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <_default />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/dashboard"
            />
            <Route
              component={[Function]}
              path="/trade/:symbol"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/dashboard"
              />,
              <Route
                component={[Function]}
                path="/trade/:symbol"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/trade/:symbol",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
